### PR TITLE
Update README docs related to local build

### LINKED ISSUE #1476: rocksdb build script fails in clang-sys with: function not loaded: clang_createIndex
I tried to compile polkadot on latest Debian sid. 

I installed `clang`, `pkg-config`, and `libssl-dev`, and then ran `cargo install --git https://github.com/paritytech/polkadot polkadot`.

That resulted in this error:

```text
error: failed to run custom build command for `librocksdb-sys v6.7.4`

Caused by:
  process didn't exit successfully: `/tmp/cargo-installhZ9LCD/release/build/librocksdb-sys-d487ffd7e62de0a3/build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-changed=build.rs

--- stderr
thread 'main' panicked at 'function not loaded: clang_createIndex', /home/josh/.cargo/registry/src/github.com-1ecc6299db9ec823/clang-sys-0.29.3/src/lib.rs:1659:1
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### PR DESCRIPTION
**Pull request by danforbes**
Closes #1476

cc @joshtriplett

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,9 +58,11 @@ rustup update
 Once done, finish installing the support software:
 
 ```bash
-sudo apt install make clang pkg-config libssl-dev
+sudo apt install make clang pkg-config libssl-dev libclang-dev
```
Comment (ordian) [inline README.md:61]: looking at https://getsubstrate.io/
> $MAKE_ME_ROOT apt install -y cmake pkg-config libssl-dev git gcc build-essential git clang libclang-dev

(note that `git` is repeated twice, also `gcc` is part of `build-essential`)

I would change it to 
```suggestion
sudo apt install build-essential git cmake clang libclang-dev pkg-config libssl-dev 
```

**Comment by danforbes** [comment]
@ordian - I will prepare a PR for this script https://github.com/paritytech/scripts/blob/master/get-substrate.sh and test it by running some commands in a Docker container. I will tag you on that PR and we can finalize that before merging in this one. How does that sound?

File: README.md [hunk 2]
```diff
@@ -29,8 +29,21 @@ cargo install --git https://github.com/paritytech/polkadot --tag <version> polka
...
-to your PATH environment variable. Restarting your computer will do this for you automatically.
+If you'd like to build from source, follow these instructions.
+
+#### Quick Setup
+
+Install all the required dependencies with a single command. (Be patient, this can take up to 30
+minutes):
+
+```bash
+curl https://getsubstrate.io -sSf | bash -s -- --fast
```
Comment (bkchr) [inline README.md:40]: Please no. Please let us kill this script finally...

Comment (danforbes) [reply README.md:40]: Can you provide more some context than that, Basti?

Comment (bkchr) [reply README.md:40]: This script was created 2018 for a demo. 

There are multiple problems with this script.

1. It installs the `substrate` node executable if you forget `--fast`, which is not required for Polkadot.
2. It only supports a small list of operating system/distributions.
3. I think it is much better to name the dependencies to make it easy to install them. I would expect that people who are using their computer, are able to install dependencies.

Comment (danforbes) [reply README.md]: Makes sense...thanks for the info. The PR has been updated.

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
Please merge master

**Comment by danforbes** [comment]
Rebased

**Pull request merged**